src/cargo/ops/cargo_add/mod.rs

```diff
@@ -0,0 +1,585 @@
+//! Core of cargo-add command
+
+mod crate_spec;
+mod dependency;
+mod manifest;
+
+use std::collections::BTreeSet;
+use std::collections::VecDeque;
+use std::path::Path;
+
+use anyhow::Context;
+use indexmap::IndexSet;
+use termcolor::Color::Green;
+use termcolor::Color::Red;
+use termcolor::ColorSpec;
+use toml_edit::Item as TomlItem;
+
+use crate::core::Registry;
+use crate::CargoResult;
+use crate::Config;
+use crate_spec::CrateSpec;
+use dependency::Dependency;
+use dependency::GitSource;
+use dependency::PathSource;
+use dependency::RegistrySource;
+use dependency::Source;
+use manifest::LocalManifest;
+
+/// Information on what dependencies should be added
+#[derive(Clone, Debug)]
+pub struct AddOptions<'a> {
+    /// Configuration information for cargo operations
+    pub config: &'a Config,
```

Comment:
1. Several other ops do this
2. While I'm not the biggest SOLID advocate, it would feel dirty / unclear violating single-responsibility by using the workspace to access process-specific configuration.  I can understand accessing that config related to the workspace or for expediency when dealing with deeply nested situations but neither applies here.